discord_bot.py

```diff
@@ -0,0 +1,323 @@
+                helpers.humanize_integer(player.xp)
+            ),
+            inline=True
+        )
+
+        embed.add_field(
+            name='Next rank',
+            value='{}\n{} XP'.format(
+                player.next_rank.name,
+                helpers.humanize_integer(player.next_rank.xp)
+            ) if player.next_rank else 'Highest possible rank reached',
+            inline=True
+        )
+
+        if player.next_rank:
+            embed.add_field(
+                name='Next rank progression',
+                value='{}% - {} XP remaining'.format(
+                    player.xp_percent_completion_to_next_rank,
+                    helpers.humanize_integer(player.xp_to_next_rank)
+                )
+            )
+
+        embed.add_field(
+            name='Kills',
+            value=helpers.humanize_integer(player.kills),
+            inline=True
+        )
+
+        embed.add_field(
+            name='Deaths',
+            value=helpers.humanize_integer(player.deaths),
+            inline=True
+        )
+
+        embed.add_field(
+            name='K/D ratio',
+            value=player.kd_ratio,
+            inline=True
+        )
+
+        embed.add_field(
+            name='Score',
+            value=helpers.humanize_integer(player.score),
+            inline=True
+        )
+
+        embed.add_field(
+            name='Time played',
+            value=helpers.humanize_seconds_to_hours(player.time_played) + ' (' + helpers.humanize_seconds_to_days(player.time_played) + ')',
+            inline=True
+        )
+
+        if player.playing_on_server:
+            embed.set_footer(text='🎮 Playing on {} ({} - {} - {}/{})'.format(
+                player.playing_on_server.name_display,
+                player.playing_on_server.type_name,
+                player.playing_on_server.map.name_display,
+                player.playing_on_server.players.current,
+                player.playing_on_server.players.max
+            ))
+
+        return embed
+
+    def create_server_message_embed(self, server, with_players_list=False):
+        """Create a RWRS server rich Discord message."""
+        embed = self.create_base_message_embed()
+
+        embed.url = server.link_absolute
+        embed.title = 'Servers › {}'.format(server.name)
+        embed.description = '[Join now]({})'.format(server.steam_join_link.replace(' ', '%20')) # FIXME Don't work
+
+        if server.website:
+            embed.description += ' • [Server website]({})'.format(server.website)
+
+        if with_players_list and server.players.list:
+            embed.add_field(
+                name='Players',
+                value=', '.join(server.players.list)
+            )
+
+        if server.map.has_preview:
+            with app.app_context():
+                embed.set_thumbnail(
+                    url=url_for('static', filename='images/maps/preview/{game_type}/{map_id}.png'.format(game_type=server.type, map_id=server.map.id), _external=True)
```

Comment:
Create a `ServerMap` string attribute with this URL. Update other places as well.